src/gcp/proto.ts

```diff
@@ -235,3 +235,30 @@ export function formatServiceAccount(serviceAccount: string, projectId: string):
   }
   return `serviceAccount:${serviceAccount}`;
 }
+
+/**
+ * Remove keys whose values are undefined.
+ * When we write an interface { foo?: number } there are three possible
+ * forms: { foo: 1 }, {}, and { foo: undefined }. The latter surprises
+ * most people and make unit test comparison flaky. This cleans that up.
+ */
+export function pruneUndefiends(obj: unknown): void {
```

Comment:
Does this have tests?

I'm really confused about what this is solving. The default value of any key that doesn't exist on an object _is_ `undefined`, so when I see below `if (keyable[key] === undefined) delete keyable[key]` I think "well, if a test was testing for the existence of something, this wouldn't have changed anything".  Maybe you can point me to a unit test that is flakey without this function?